### ISSUE TITLE
Error installing on Windows: "Path must be a string. Received undefined"

### ISSUE THREAD

**Issue by alistairmgreen** [npm/npm-merge-driver#5]
The installation command from the README file produces errors on Windows 7:

```text
>  npx npm-merge-driver install --global
npx: installed 1 in 2.163s
Path must be a string. Received undefined
npx: installed 52 in 7.412s
C:\Users\greena2\AppData\Roaming\npm-cache\_npx\6568\node_modules\npm-merge-driver\index.js
Path must be a string. Received undefined
```

#### Additional information
* npm version 5.8.0
* node.js version 8.11.1
* Terminal is PowerShell 5.1
* git version 2.16.1.windows.4

**Comment by andrewhosgood** [comment]
Someone in my team had this issue and it was because there was a space in their $PATH

**Comment by Teamop** [comment]
Got same error with npm v6.1.0 on win10, but if just install locally `npx npm-merge-driver install`, successfully

**Comment by VoldemLive** [comment]
here I was found solution: https://blog.yinaoxiong.cn/2018/08/19/fix-npx-erro.html
Hope this will help you)

**Comment by mybluedog24** [comment]
Thanks to VoldemLive's solution:
1. run: `npm i -g npx` upgrade npx to the newest version
2. go to Control Panel -> System and Security -> System -> Advanced system settings -> Environment Variables
3. copy the path for npx from user variables to system variables, the path should be like `C:\Users\{user name}\AppData\Roaming\npm`. Make sure this PATH is before the nodejs PATH in system variable.

**Comment by GusGusGusGus** [comment]
@mybluedog24  should't we refrain from installing global packages due to potential version mismatch in projects and lack of project self-containing? ( https://codeburst.io/maybe-dont-globally-install-that-node-js-package-f1ea20f94a00 )

**Comment by GusGusGusGus** [comment]
I tried to replicate your steps and neither installing npx locally nor globally registers an Environment Variable on my Windows. I do have an _npm_ env variable on the Roaming folder. Upon inspection, that folder does have an npx text file and a .cmd file. Besides that, on the node_modules folder there is also an npx folder with a bunch of stuff. I keep getting the 'Path must be string error. Received undefined', which I suppose refers to that exact variable.

**Comment by mybluedog24** [comment]
@GusGusGusGus Yes, it's better to be installed locally. I forgot why I installed it globally. Have you tried run `npx -v` in your project folder and outside of project folder? If they are different versions then you haven't set the variable properly.

In my user variable I have this:
`C:\Users\Frank\AppData\Roaming\npm`
and I copy it to global variables before `nodejs`:

```
C:\Users\Frank\AppData\Roaming\npm
C:\Program Files\nodejs\
```

And then everything works. Hope this helps.

**Comment by GusGusGusGus** [comment]
Yes it did, thank you. When you said:

> 3\. the path should be _like_ `C:\Users\{user name}\AppData\Roaming\npm`

I thought you meant the npx folder inside npm's folder (simply because it was the only one I could find).
I followed your steps and that solved the "path must be a string, received undefined" error. Npx should warn about this on their instructions.

**Comment by aemooooon** [comment]
> here I was found solution: https://blog.yinaoxiong.cn/2018/08/19/fix-npx-erro.html
> Hope this will help you)

中文的，非常感谢。已解决

**Comment by Jack-Works** [comment]
Run it in PowerShell not cmd
